pkg/apis/config/v1alpha1/types.go

```diff
@@ -78,6 +81,9 @@ type BalloonsPolicySpec struct {
 	Log log.Config `json:"log,omitempty"`
 	// +optional
 	Instrumentation instrumentation.Config `json:"instrumentation,omitempty"`
+	// +optional
+	// +kubebuilder:default={"nodeResourceTopology": true }
+	Agent AgentConfig `json:"agent,omitempty"`
```

Comment:
Wouldn't adding the policy-agnostic agent configuration under each policy CR lead to conflicts when multiple policies attempt to set different values for the same fields, such as `nodeResourceTopology`?

Reply:
I'm not sure what you mean. This newly added agent-specific configuration value configures the behavior of the agent (wrt. whether the the Node Resource Topology and Pod Resource API functionality is made available) for each policy. It is similar to the other policy-agnostic configuration bits, for instance instrumentation. Everything in there is purely an input there for the resource manager and other infra bits, policies never set there anything. Multiple policies cannot conflict, because they have the same piece of data in one of their own policy-specific configuration CR, in some instance of `ballonspolicies.config.nri`, `topologyawarepolicies.config.nri` or `templatepolicies.config.nri`.

Reply:
What i meant is that you are allowing the agent configuration to be modified by all all the policies (well not all but more than one). Meaning, when more than one policy is installed a user can disable NRT from one policy and enable from the other. How do you handle that?

Reply:
Sure, but that is deliberately done so by design. I mean... how is this different from, for instance, instrumentation being enabled for one policy but not for another ?

Reply:
> Meaning, when more than one policy is installed, a user can disable NRT from one policy and enable from the other. How do you handle that?

Well, those policy instances for which NRT is disabled in the agent will not be exporting NRT CRs, and those instance for which NRT is enabled in the agent will be exporting NRT CRs.

Reply:
Sorry, I had misunderstanding of the agent instances. We can close this chat.